fenix/app/src/main/java/org/mozilla/fenix/shopping/state/ReviewQualityCheckAction.kt

```diff
@@ -9,4 +9,47 @@ import mozilla.components.lib.state.Action
 /**
  * Actions for review quality check feature.
  */
-sealed interface ReviewQualityCheckAction : Action
+sealed interface ReviewQualityCheckAction : Action {
+
+    /**
+     * Actions that are observed by middlewares.
+     */
+    sealed interface MiddlewareAction : ReviewQualityCheckAction
+
+    /**
+     * Actions that cause updates to state.
+     */
+    sealed interface UpdateAction : ReviewQualityCheckAction
+
+    /**
+     * Actions related to preferences.
+     */
+    sealed interface PreferencesMiddlewareAction : MiddlewareAction
+
+    /**
+     * Triggered when the store is initialized.
+     */
+    object Init : PreferencesMiddlewareAction
+
+    /**
+     * Triggered when the user has opted in or out of the review quality check feature.
+     */
+    data class OptInAction(
+        val hasUserOptedIn: Boolean,
+    ) : PreferencesMiddlewareAction
+
+    /**
+     * Triggered when the user has enabled or disabled product recommendations.
+     */
+    data class ProductRecommendationAction(
```

Comment:
This might make the verbiage and logic flow a little more clear upstream and downstream. We have existing examples of using two actions to toggle on/off part of the state vs one action with a boolean.
```suggestion
    object EnableProductRecommendation(
    object DisableProductRecommendation(
```

Reply:
Alternatively, use something like `ToggleProductRecommendations` and let the reducer and middleware change it based on previous state.